saist/util/rules.py

```diff
@@ -0,0 +1,39 @@
+import os
+import yaml
+
+class PromptRules:
+    RulesFile = "saist.rules"
+
+    @staticmethod
+    def apply_rules(prompt):
+        rules = PromptRules.load_rules()
+
+        override_prompt = rules.get("PROMPT_OVERRIDE")
+        if override_prompt:
+            #skip pre and post if override is present
+            return override_prompt
+
+        pre = rules.get("PROMPT_PRE", "")
+        post = rules.get("PROMPT_POST", "")
+
+        #add pre/post if they are present
+        return f"{pre}{prompt}{post}"
+
+    @staticmethod
+    def load_rules():
+        if not os.path.exists(PromptRules.RulesFile):
+            print("No saist.rules file found.")
```

Comment:
Take a look at how we do logging in the codebase and amend these print statements. Our logging is pretty poorly standardised, but yo should find a correct way of doing it